impl/task_runner_do.go

```diff
@@ -86,6 +87,24 @@ func (d *DoTaskRunner) runTasks(input interface{}, tasks *model.TaskList) (outpu
 		}
 
 		d.TaskSupport.SetTaskStatus(currentTask.Key, ctx.PendingStatus)
+
+		// Check if this task is a SwitchTask and handle it
```

Comment:
The logic for handling the next task to execute is duplicated here and in task_runner_do, can it be located in a helper method to avoid duplication?

Reply:
I don't get it, wdym? This is the `task_runner_do` file; I can see this logic is only being handled here. 🤔